pkg/scanners/scannerv4/scannerv4.go

```diff
@@ -202,3 +208,92 @@ func (s *scannerv4) GetVulnerabilities(image *storage.Image, components *types.S
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), scanTimeout)
+	defer cancel()
+
+	vr, err := s.scannerClient.GetVulnerabilities(ctx, nodedigest, indexReport.GetContents())
+	if err != nil {
+		log.Errorf("Failed to create vulnerability report: %v", err)
+		return nil, errors.Wrap(err, "Failed to create vulnerability report")
+	}
+
+	return vr, nil
+}
+
+func (s *scannerv4) GetNodeInventoryScan(node *storage.Node, inv *storage.NodeInventory, ir *v4.IndexReport) (*storage.NodeScan, error) {
+	if ir == nil && inv != nil {
+		return nil, errors.New("Received Scanner v2 data for Scanner v4. Exiting.")
+	}
+	vr, err := s.GetNodeVulnerabilityReport(node, ir)
+	if err != nil {
+		log.Errorf("Failed to create vulnerability report: %v", err)
+		return nil, errors.Wrap(err, "Failed to create vulnerability report")
+	}
+	log.Infof("Received Vulnerability Report with %d vulnerabilities: %v", len(vr.Vulnerabilities), vr)
+	// TODO(ROX-26089): Convert resulting VulnerabilityReport to storage.NodeScan and return that
+	return nil, nil
+}
+
+func (s *scannerv4) GetNodeScan(_ *storage.Node) (*storage.NodeScan, error) {
+	return nil, errors.New("Not implemented for Scanner v4")
+}
+
+func (s *scannerv4) TestNodeScanner() error {
+	log.Warn("NodeScanner v4 - Returning FAKE 'success' to Test")
+	return nil
+}
+
+// NodeScannerCreator provides the type scanners.NodeScannerCreator to add to the scanners registry.
+func NodeScannerCreator() (string, func(integration *storage.NodeIntegration) (scannerTypes.NodeScanner, error)) {
+	return scannerTypes.ScannerV4, func(integration *storage.NodeIntegration) (scannerTypes.NodeScanner, error) {
+		return newNodeScanner(integration)
+	}
+}
+
+func newNodeScanner(integration *storage.NodeIntegration) (*scannerv4, error) {
+	conf := integration.GetScannerv4()
+	if conf == nil {
+		return nil, errors.New("scanner v4 configuration required")
+	}
+	indexerEndpoint := DefaultIndexerEndpoint
+	if conf.IndexerEndpoint != "" {
+		indexerEndpoint = conf.IndexerEndpoint
+	}
+
+	matcherEndpoint := DefaultMatcherEndpoint
+	if conf.MatcherEndpoint != "" {
+		matcherEndpoint = conf.MatcherEndpoint
+	}
+
+	numConcurrentScans := defaultMaxConcurrentScans
+	if conf.GetNumConcurrentScans() > 0 {
+		numConcurrentScans = int64(conf.GetNumConcurrentScans())
+	}
+
+	log.Debugf("Creating Scanner V4 with name [%s] indexer address [%s], matcher address [%s], num concurrent scans [%d]", integration.GetName(), indexerEndpoint, matcherEndpoint, numConcurrentScans)
+	ctx := context.Background()
+	c, err := client.NewGRPCScanner(ctx,
+		client.WithIndexerAddress(indexerEndpoint),
+		client.WithMatcherAddress(matcherEndpoint),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	scanner := &scannerv4{
+		name:              integration.GetName(),
+		activeRegistries:  nil,
+		ScanSemaphore:     types.NewSemaphoreWithValue(numConcurrentScans),
+		NodeScanSemaphore: types.NewNodeSemaphoreWithValue(numConcurrentScans), // TODO: Set custom value
```

Comment:
Same here, let's add a new number, set it to something reasonable, and call this TODO done.

Reply:
let's address that one